### DISCUSSION TITLE
File not found

### DISCUSSION THREAD

**Discussion by esp0314** [thunderclient/thunder-client-support#1384] [Q&A]
Hi,

I have a custom js script wherein it reads a text file and check the value. But when run in in CLI, it says file not found. The location of the envi.txt is the same location with my custom js script. Can you please help me?
const filePath = 'tests/envi.txt';
const endPoint1 = "https://www.google.com/abc";
const endPoint2 = "https://www.yahoo.com/abc";
var env = await tc.readFile(filePath);
console.log("Environment: " + env);
if (env.toLowerCase() == "beta") {
        tc.setVar("endPoint", endPoint1, "local");
        await tc.runRequest("id");
    } else {
        tc.setVar("endPoint", endPoint2, "local");
        await tc.runRequest("id");
    }

also tried const filePath = './tests/envi.txt'; but cannot find also the txt file

**Comment by rangav** [comment]
Hi @esp0314 thanks for reporting will look into it.

can you please verify subscription by using below form, needed for support.
https://www.thunderclient.com/contact

**Reply by esp0314** [reply]
Done. I am paid version. Thank you

**Reply by rangav** [reply]
Thanks. Will verify and get back to your question shortly.

**Comment by rangav** [comment]
You need to use relative path to the project/workspace
- right click on the file -> then select `Copy Relative Path` ( see below image)
<img width="286" alt="Screenshot 2023-10-25 at 09 27 21" src="https://github.com/rangav/thunder-client-support/assets/8637550/7d68a04f-86fe-4ac7-b737-a2648fda8afe">

**Reply by esp0314** [reply]
I did but still the same

**Reply by rangav** [reply]
Can you please run `tc --debug` and share screenshot

**Comment by esp0314** [comment]
![image](https://github.com/rangav/thunder-client-support/assets/148940089/ba59c79d-380f-491c-ab8b-118492db721d)

i only have issue "File Not Found" on the CLI, but when run in the UI it works fine

**Reply by rangav** [reply]
can you remove everything and run this command  `tc --debug`, remove --ws and all other data

**Reply by esp0314** [reply]
Here it is
![image](https://github.com/rangav/thunder-client-support/assets/148940089/57587424-9aba-4585-abf0-e45729eb3725)

**Reply by rangav** [reply]
thanks, run this command - `tc dc-get-periods-kbb` and remove all other data

**Reply by esp0314** [reply]
it works. No error now.
ran the whole collections
![image](https://github.com/rangav/thunder-client-support/assets/148940089/e9ea9490-b534-442b-8f36-7d79d2f05295)

**Reply by rangav** [reply]
thanks for confirmation. you dont need to use `--ws` switch as you already running commands from root of your project.

**Reply by esp0314** [reply]
the reason i used the --ws is i need to point to the collections(dc, dcmobile, uw) because i am running this in azure pipeline
![image](https://github.com/rangav/thunder-client-support/assets/148940089/c8badd81-0a2c-4ee2-96f6-eaea535836f1)

**Reply by rangav** [reply]
dcmobile is collection name? Are you sure? 
- run this command `tc list` to see all collection names

**Reply by esp0314** [reply]
as you can see, i have 3 folders(dc, dcmobile, uw). Those are collections. For me to work on the collection, i have to edit the settings.json and point to the collection that i will work on. Now, in order to make this work, i have to edit the settings.json and run the command line
![image](https://github.com/rangav/thunder-client-support/assets/148940089/840de091-64c9-4885-be34-8954c3d6d66e)

![image](https://github.com/rangav/thunder-client-support/assets/148940089/c69f07ec-3a34-4f14-b7fa-a0c97fee5884)


![image](https://github.com/rangav/thunder-client-support/assets/148940089/d3c2bdf3-c130-4ec6-a65a-8848087c3c7f)


Now, in the pipeline, i dont have a way to edit the settings

**Reply by rangav** [reply]
Ok understood your use cases

in that case try use `--ws` but use full path instead of relative path.

**Comment by esp0314** [comment]
this is the full path, but in the azure pipeline agent, this will be different, right? Then it will not work
C:\BookServices\Nowcom.Services.BookingNetCore\api-tests\dc
C:\BookServices\Nowcom.Services.BookingNetCore\api-tests\dcmobile
C:\BookServices\Nowcom.Services.BookingNetCore\api-tests\uw

**Reply by rangav** [reply]
Ok got it. will make changes to `--ws` to accept full or relative path

**Reply by esp0314** [reply]
Thank you. But then again, will this resolve my issue that i can run a collection without editing the settings.json? Like the screenshot below?
![image](https://github.com/rangav/thunder-client-support/assets/148940089/fde6e30f-9407-44f8-862b-d2700fe562fa)

**Reply by rangav** [reply]
Yes I  think it will solve the issue. When its implemented I will test and let you know

**Reply by esp0314** [reply]
Thanks, muchly.

**Reply by rangav** [reply]
Why dont you create multiple collections in one folder, instead of 3 folders? Will it not work for you?

**Reply by esp0314** [reply]
Reason i created multiple collections is, i am not the only one working on this. If one collection, the tendency is we are editing the same file and it will cause merge conflicts

**Reply by rangav** [reply]
I have published new version, please update CLI to `v1.8.3`

- `--ws` now supports relative or absolute path

Please test and let me know your feedback

**Reply by esp0314** [reply]
Still, file not found

**Reply by rangav** [reply]
can we do a call, to understand the issue?

**Reply by esp0314** [reply]
yeah, lets have a call.

**Reply by rangav** [reply]
will send you zoom link in few minutes

**Reply by rangav** [reply]
sent you invite

**Comment by rangav** [comment]
I have fixed the issue, Please update CLI to `v1.8.4`

Please test and let me know

**Reply by esp0314** [reply]
so far looks good, ill let you know if i encounter any issue. Thank you for attending this matter

**Reply by rangav** [reply]
Thanks for confirmation

**Reply by esp0314** [reply]
It looks like there is still an issue with the latest version(1.8.4).
As you can see on the 1st screenshot, there were no results but created raw logs in the pipeline run
![image](https://github.com/rangav/thunder-client-support/assets/148940089/b96a2b32-0bea-4a6e-998e-3491d96688ac)

![image](https://github.com/rangav/thunder-client-support/assets/148940089/1b141f02-029c-4bce-8c92-137328b6385d)

![image](https://github.com/rangav/thunder-client-support/assets/148940089/8ed77e9d-f5cd-4958-b488-7b3fbd562b3d)

Now, i reverted to version 1.8.1, and i was able to get a results and there were no errors in the azure pipeline
![image](https://github.com/rangav/thunder-client-support/assets/148940089/eadc26ae-5984-4d20-98fa-3c27d31c062d)


![image](https://github.com/rangav/thunder-client-support/assets/148940089/2a7f7b7c-4f88-45ed-85c6-dd5571813ece)


![image](https://github.com/rangav/thunder-client-support/assets/148940089/f48d50eb-c53f-4a66-8948-df5b21dda8a8)

**Reply by rangav** [reply]
Thanks for sharing info about the issue.

what command did you run in CI/CD? Can you please share details

**Reply by esp0314** [reply]
this is the command in CICD
const tcProcess = exec(`tc --ws api-tests/dc --col dc --report xml --env test-data-int0 || exit /b 0`)

**Reply by rangav** [reply]
Thanks for sharing info, will verify and get back to you

**Reply by rangav** [reply]
I have tested locally its working for me

Can you test locally and generate xml report with `v1.8.4`? just want to verify if is it happening in CI/CD only.

**Reply by esp0314** [reply]
yes, it is only happening in CICD. I tested locally and it works fine
![image](https://github.com/rangav/thunder-client-support/assets/148940089/ca81a805-86de-42b9-adec-7506fe4e68dd)

![image](https://github.com/rangav/thunder-client-support/assets/148940089/870dbc19-a845-444c-9a8c-051feb2795e1)

**Reply by esp0314** [reply]
one thing i also noticed is that, the xml content for new version(1.8.4) of CLI is different with the older version. Not sure if has something to do with the issue

**Reply by rangav** [reply]
Yeah updated xml format in v1.8.4

Can you please test by removing report xml and use json and html in CI/CD

`--report "json,html"`

**Reply by esp0314** [reply]
I think in CICD azure pipeline, JUnits accepts xml based test reports, right?

**Reply by rangav** [reply]
dont know I am not an expert in this junit or xml reports. May be search in google

**Reply by esp0314** [reply]
yes, in our CICD we only used xml

**Reply by rangav** [reply]
Ok got it

can you add `cli` also and see if the requests are executed or not

`--report "cli,xml"`

Just want to find the issue is with generating xml report path or other issue.

**Reply by esp0314** [reply]
yes, the request is executed and xml is also generated

**Reply by rangav** [reply]
the above screenshot is using version `v1.8.4`?

**Reply by esp0314** [reply]
yes, its 1.8.4

**Reply by rangav** [reply]
thats good, you got it working then? what was the issue earlier?

**Reply by esp0314** [reply]
if you go back to the previous thread, i said that "xml is generated" using 1.8.4 but it is not displaying the results in the azure pipeline. When i reverted to 1.8.1, results are displayed. The only changes is the package json. Go back to this thread
![image](https://github.com/rangav/thunder-client-support/assets/148940089/99060175-49e8-4f2a-88a9-32a75cccc263)

**Reply by rangav** [reply]
Thanks for confirmation.

if you need any changes to xml format let me know.

**Reply by esp0314** [reply]
thank you
